design-proposals/primary_network_binding_for_seamless_migration.md

```diff
@@ -0,0 +1,520 @@
+| Pod Probes Support          | Yes                                                                                                             | No                                   | No                                               |
+| Seamless Live-Migration     | Requires Development                                                                                            | Yes                                  | Yes                                              |
+| Supported Traffic           | TCP/UDP/ICMP (All IP in case seamless migration is resolved)                                                    | IP                                   | IP                                               |
+| Migration Support Indicator | Yes                                                                                                             | Requires Development                 | Yes                                              |
+| Mesh / Istio support        | Yes                                                                                                             | No                                   | No                                               |
+| IP Status (reporting)       | Yes                                                                                                             | Yes                                  | Requires Development                             |
+| Observability Integration   | Yes                                                                                                             | No                                   | No                                               |
+| Resource Overhead           | 500MB per VM, 250MB per VM already available on U/S, Negligible overhead when resolving the seamless migration. | Negligible                           | Negligible                                       |
+| Performance                 | N/A                                                                                                             | N/A                                  | N/A , Most likely better than the bridge option. |
+| Kubevirt Support            | Low                                                                                                             | High                                 | Medium                                           |
+
+[*] Core Bridge binding can be deployed as a binding plugin with a subset
+    of its existing feature sets. E.g. as a plugin, IP & routes can be
+    supported only through the network-provider.
+
+### The Binding Winner
+
+Considering the requirements, development effort and maintenance cost,
+the following plan is chosen:
+1. Develop a **macvtap** network binding plugin.
+2. In case option [1] fails, develop a **bridge** network binding plugin.
+3. In case option [2] fails, enhance the existing **bridge** core binding.
```

Comment:
It is really not clear to me why macvtap is preferred to bridge, especially when deployed as a plugin.

The order of the priorities here is not what I would expect to be honest, after going through the table above.

Reply:
IMO the bridge one has lower performance compared to the macvtap passthru.
If we did not encounter the problems with creating a macvtap through the CNI, we would even had this out-of-tree, reducing the maintenance cost.

Even now, after discovering all the problem with cgroups and the yet unclear reason it does not work for ingress traffic, I would have not prioritize this differently.

Anyway, as the current attempt failed with macvtap and the realization that cgroup manipulation is really problematic in Kubevirt today, we ended up with choosing the option of the bridge plugin (which seems to work as expected).